avaframe/ana3AIMEC/aimecTools.py

```diff
@@ -119,67 +121,219 @@ def fetchReferenceSimNo(avaDir, inputsDF, comModule, cfgSetup):
-            if typeCP == str:
-                sortingValues = [x.lower() for x in sortingParameter]
-                indexRef = sortingValues.index(typeCP(cfgSetup['referenceSimValue'].lower()))
-                valRef = sortingParameter[indexRef]
-            elif typeCP in [float, int]:
-                colorValues = np.asarray(sortingParameter)
-                indexRef = (np.abs(colorValues - typeCP(cfgSetup['referenceSimValue']))).argmin()
-                valRef = colorValues[indexRef]
-            else:
-                indexRef = sortingParameter.index(typeCP(cfgSetup['referenceSimValue']))
-                valRef = sortingParameter[indexRef]
-            log.info('Reference Simulation is based on %s = %s - closest value found is: %s' %
-                     (varParList[0], cfgSetup['referenceSimValue'], str(valRef)))
-            refSimulation = inputsDF[inputsDF[varParList[0]] == valRef]['simName'].to_list()[0]
-            colorVariation = True
+    # if the simulations come from com1DFA, it is possible to order the files and define a reference
+    if configFound and cfgSetup['varParList'] != '':
+        # fetch parameters that shall be used for ordering
+        varParList = cfgSetup['varParList'].split('|')
+        # order simulations
+        varParList, inputsDF = cfgUtils.orderSimulations(varParList, cfgSetup.getboolean('ascendingOrder'), inputsDF)
+        # now look for the reference
+        if referenceSimValue != '':
+            # if a referenceSimValue is provided, find the corresponding simulation
+            # get the value of the first parameter used for ordering (this will be usefull for colorcoding in plots)
+            refSimRowHash, refSimName = defineRefOnSimValue(referenceSimValue, varParList, inputsDF)
+
+        elif cfgSetup['referenceSimName'] != '':
+            # else if a referenceSimName is provided, find the corresponding simulation - set
+            # simulation with referenceSimName in name as referene simulation
+            refSimRowHash, refSimName = defineRefOnSimName(referenceSimName, inputsDF)
         else:
+            # if no referenceSimValue is provided, we assume the first simulation after reordering is the reference
             # reference simulation
-            refSimulation = inputsDF.iloc[0]['simName']  # inputsDF.head(1)['simName'].values[0]
-            colorVariation = False
-
-    elif cfgSetup['referenceSimName'] != '':
-        colorVariation = False
-        # if no colorVariation info and refeenceSimValue available but referenceSimName is given - set
+            refSimRowHash = inputsDF.index[0]
+            refSimName = inputsDF.loc[refSimRowHash, 'simName']
+            log.info(('Reference simulation is the first simulation after reordering according to %s in ascending order = %s and corresponds to simulation %s')
+                     % (varParList, cfgSetup.getboolean('ascendingOrder'), refSimName))
+        colorVariation = True
```

Comment:
This is the correct spot, here we reordered the simulations because 'varParList' is provided but there is no 'referenceSimName' or 'referenceSimValue' so we simply take the first sim after reordering.